### PR TITLE
Attach Dialog improvements

### PR DESCRIPTION
**Pull request by Forsari0**
## Summary of PR
- Add **Show PID in HEX** setting in **GUI** subtab (affects only **Attach** dialog, default is **TRUE**)
-  Add new column in **Attach** dialog - **CommandLine arguments**, which is show command line arguments taken from target process **PEB** structure

For demonstration purposes, here it is screenshot:
x32dbg_2016-08-20_05-11-23

---
This change is [Reviewable](https://reviewable.io/reviews/x64dbg/x64dbg/994)

### CODE REVIEW

File: src/dbg/_dbgfunctions.cpp [hunk 1]
```diff
@@ -163,17 +163,21 @@ static bool _getjit(char* jit, bool jit64)
...
+    *count = (int)infoList.size();
     if(!*count)
         return false;
     *entries = (DBGPROCESSINFO*)BridgeAlloc(*count * sizeof(DBGPROCESSINFO));
     for(int i = 0; i < *count; i++)
     {
-        (*entries)[*count - i - 1].dwProcessId = list.at(i).th32ProcessID;
-        strcpy_s((*entries)[*count - i - 1].szExeFile, list.at(i).szExeFile);
+        (*entries)[*count - i - 1].dwProcessId = infoList.at(i).th32ProcessID;
+        strncpy((*entries)[*count - i - 1].szExeFile, infoList.at(i).szExeFile, MAX_PATH);
```
Comment (mrexodia) [inline src/dbg/_dbgfunctions.cpp]: instead of the manual `'\0'` method you could use `strncpy_s(szExeFile, infoList.szExeFile, _TRUNCATE)`

**Comment by mrexodia** [comment]
Sorry for the delay, I have been very ill recently and I cannot really do much at the moment.

File: src/dbg/debugger.cpp [hunk 2]
```diff
@@ -2116,39 +2202,66 @@ bool dbgsetcmdline(const char* cmd_line, cmdline_error_t* cmd_line_error)
...
-    }
+        duint NumberOfBytesRead;
+        if(!MemoryReadSafe(hProcess, (LPVOID)cmd_line_error->addr, &CommandLine, sizeof(UNICODE_STRING), &NumberOfBytesRead))
+        {
+            cmd_line_error->type = CMDL_ERR_READ_GETCOMMANDLINEBASE;
+            return false;
+        }
 
-    Memory<wchar_t*> wstr_cmd(CommandLine.Length + sizeof(wchar_t));
+        wstr_cmd = new Memory<wchar_t*>(CommandLine.Length + sizeof(wchar_t));
```
Comment (mrexodia) [inline src/dbg/debugger.cpp]: This appears to be incorrect. I believe you should use a *C++* style declaration (eg not change this line at all and declare the variable only once its needed).

File: src/dbg/debugger.cpp [hunk 3]
```diff
@@ -2116,39 +2202,66 @@ bool dbgsetcmdline(const char* cmd_line, cmdline_error_t* cmd_line_error)
...
     }
+    else
+    {
+        if(!MemRead(cmd_line_error->addr, &CommandLine, sizeof(CommandLine)))
+        {
+            cmd_line_error->type = CMDL_ERR_READ_PROCPARM_PTR;
+            return false;
+        }
+
+        wstr_cmd = new Memory<wchar_t*>(CommandLine.Length + sizeof(wchar_t));
```
Comment (mrexodia) [inline src/dbg/debugger.cpp]: If you want to resize the memory, you can use `wstr_cmd.resize()`

**Comment by mrexodia** [comment]
Thanks for your contribution! Merged in around f09586ea

Also #884 might be interesting since you wrote some command line utility stuff...

**Comment by Forsari0** [comment]
Thanks.
I'll look into #884 right after next PR.

**Pull request closed without merging**